File: ArtikullAndroid/app/src/main/java/com/example/artikujt/models/PaginatedList.java
package com.example.artikujt.models;

import java.util.ArrayList;

public class PaginatedList<T> {
    public ArrayList<T> models;
    public int numberOfPages;
    public ArrayList<Page> pages;

    public PaginatedList(ArrayList<T> _models, int _numberOfPages, int currentPageIndex){
        models = _models;
        numberOfPages = _numberOfPages;

        for (int i = 1; i <= numberOfPages; i++)
        {
            pages.add(new Page(i, i == currentPageIndex));
        }
    }

    public class Page
    {
        public int index;
        public boolean isActive;

        public Page(int _index, boolean _isActive)
        {
            index = _index;
            isActive = _isActive;
        }
    }
}

File: ArtikujtClient/ArtikujtClient/Models/PaginatedList.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.Eventing.Reader;
using System.Linq;
using System.Web;

namespace ArtikutClient.Models
{
    public class PaginatedList<T>
    {
        public IEnumerable<T> Models { get; }
        public int NumberOfPages { get; }

        public IList<Page> Pages { get; } = new List<Page>();

        public PaginatedList(IEnumerable<T> models, int numberOfPages, int currentPageIndex)
        {
            Models = models;
            NumberOfPages = numberOfPages;

            for (int i = 1; i <= NumberOfPages; i++)
            {
                Pages.Add(new Page(i, i == currentPageIndex));
            }
        }
    }

    public struct Page
    {
        public int Index { get; }
        public bool IsActive { get; }

        public Page(int index, bool isActive)
        {
            Index = index;
            IsActive = isActive;
        }
    }


}